### PR TITLE
Remove unused code

### LINKED ISSUE #695: Dead code in Library and mock_helper for pseudo functions
We add a pseudo function to API metadata in order to generate methods for "Dispatcher". For example, src/nifgen/functions_add_on.py adds WriteWaveformDispatcher. This doesn't correspond to any functions in the DLL, rather it's used in order to inject our own templates that take care of dispatching the call to the real function.

However, code for calling the DLL and for mocking that call is still generated. This is non-functional confusing dead code. We should skip generating them in Library and mock_helper.

### PR DESCRIPTION
**Pull request by texasaggie97**
- [X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
- [X] ~~I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~~
- [X] ~~I've added tests applicable for this pull request~~
### What does this Pull Request accomplish?
* Removes functions that don't exist in the driver from library.py and mock_helper.py
  * They are dead code and will never be used
* Support additional `'codegen_method'` value of `'api'`
  * Added filter method that filters out `'api'` and `'no'` values
* Updated nidcpower, nidmm, nifake, nifgen, and niscope metadata
* No user visible change
### List issues fixed by this Pull Request below, if any.
* Fix #695 
### What testing has been done?
* Unit
* System
* Inspection of generated files

### CODE REVIEW

File: build/helper/metadata_filters.py [hunk 1]
```diff
@@ -301,6 +301,11 @@ def filter_codegen_functions(functions):
     return {k: v for k, v in functions.items() if v['codegen_method'] != 'no'}
 
 
+def filter_library_functions(functions):
+    '''Returns function metadata only for those functions to be included in codegen'''
```
Comment (marcoskirsch) [inline build/helper/metadata_filters.py]: A little confusing, because we do code generate for the 'api' only functions. Just not in the library.
Maybe reword: Returns function metadata only for those functions we want to include in the Library class. ?

File: docs/nidcpower/session.rst [hunk 2]
```diff
@@ -328,14 +328,10 @@ nidcpower.Session
    +------------------------------------------------------------+
    | :py:func:`get_channel_name`                                |
    +------------------------------------------------------------+
-   | :py:func:`get_ext_cal_last_date_and_time`                  |
```
Comment (marcoskirsch) [inline docs/nidcpower/session.rst]: We shouldn't remove these, they work without cal session and they are useful in user programs.

File: docs/niscope/functions.rst [hunk 3]
```diff
@@ -1191,238 +1191,6 @@ niscope.Session methods
 
     :type signal_identifier: str
 
-.. py:method:: fetch(num_samples=None, relative_to=niscope.FetchRelativeTo.PRETRIGGER, offset=0, record_number=0, num_records=None, timeout=datetime.timedelta(seconds=5.0))
```
Comment (marcoskirsch) [inline docs/niscope/functions.rst]: Why is this disappearing from docs?

File: generated/nidcpower/library.py [hunk 4]
```diff
@@ -263,22 +261,6 @@ def niDCPower_GetExtCalRecommendedInterval(self, vi, months):  # noqa: N802
                 self.niDCPower_GetExtCalRecommendedInterval_cfunc.restype = ViStatus  # noqa: F405
         return self.niDCPower_GetExtCalRecommendedInterval_cfunc(vi, months)
 
-    def niDCPower_GetLastExtCalLastDateAndTime(self, vi, month):  # noqa: N802
```
Comment (marcoskirsch) [inline generated/nidcpower/library.py]: I wonder if we should call all these functions "fancy whatever" so it's obvious they are only fancy python wrapping functions.

File: generated/nifgen/library.py [hunk 5]
```diff
@@ -37,7 +37,6 @@ def __init__(self, ctypes_library):
         self.niFgen_CreateAdvancedArbSequence_cfunc = None
         self.niFgen_CreateArbSequence_cfunc = None
         self.niFgen_CreateFreqList_cfunc = None
-        self.niFgen_CreateWaveformDispatcher_cfunc = None
```
Comment (marcoskirsch) [inline generated/nifgen/library.py]: ... for example we could call this niFgen_FancyCreateWaveform

File: src/nidcpower/metadata/functions_addon.py [hunk 6]
```diff
@@ -149,7 +149,7 @@
     },
     # Public function that wraps driver function but returns datetime object instead of individual items
     'GetLastExtCalLastDateAndTime': {
-        'codegen_method': 'public',
+        'codegen_method': 'api',
```
Comment (marcoskirsch) [inline src/nidcpower/metadata/functions_addon.py]: I think `python-only` would be more self-explanatory. `api` doesn't make it clear if it's the Python bindings or the C API.

Also: Don't forget to update the wiki once this is merged!

File: build/helper/metadata_filters.py (hunk 1 above)
Comment (texasaggie97) [reply build/helper/metadata_filters.py]: Add clarification text: `Returns function metadata only for those functions to included the library layer (library.py and mock_helper.py)`

File: docs/nidcpower/session.rst (hunk 2 above)
Comment (texasaggie97) [reply docs/nidcpower/session.rst]: Missed a conditional. Added back.

File: docs/niscope/functions.rst (hunk 3 above)
Comment (texasaggie97) [reply docs/niscope/functions.rst]: Missed conditional. Not purposeful.

File: generated/nidcpower/library.py (hunk 4 above)
Comment (texasaggie97) [reply generated/nidcpower/library.py]: In this PR?

File: src/nidcpower/metadata/functions_addon.py (hunk 6 above)
Comment (texasaggie97) [reply src/nidcpower/metadata/functions_addon.py]: Changed to `'python-only'`

File: generated/nidcpower/library.py (hunk 4 above)
Comment (marcoskirsch) [reply generated/nidcpower/library.py]: no way

File: docs/nidcpower/session.rst (hunk 2 above)
Comment (marcoskirsch) [reply docs/nidcpower/session.rst]: The diff still doesn't show them...

Comment (texasaggie97) [reply docs/nidcpower/session.rst]: There was yet another place where filtering was happening.

Added `filter_public_functions()` and used that for documentation.
